Powersync environment

Dieterbe opened the pull request.

Dieterbe committed:
add dev-postgres-powersync environment

Dieterbe committed:
add more startup options

rolandgeider committed:
Add a test powersync key pair

rolandgeider committed:
Update JWT powersync keys

Dieterbe committed:
fixes/tweaks

rolandgeider committed:
Build pg_ivm postgres extension to be used with powersync

rolandgeider committed:
Simplify powersync docker-compose.yml file

Now the .env file only has local configurations like the path to the wger
backend checkout

Dieterbe commented on postgres/Dockerfile:
```diff
@@ -0,0 +1,24 @@
+FROM postgres:15-bookworm AS builder
```
Why do we need our dockerfile for postgres ?

rolandgeider replied:
We need it to build the ivm extension

Dieterbe commented on dev-postgres-powersync/docker-compose.yml:
```diff
@@ -55,7 +58,11 @@ services:
+
+networks:
+  default:
+    name: wger-dev-network
```
why are we re-introducing the network override? didn't we just remove this? and what does it have to do with pg_ivm?

rolandgeider replied:
it has nothing to do with it, but I wanted to keep the dev services separate, otherwise it's not possible to run both the production and the dev stack at the same time since they would both access the same db, etc.

Dieterbe commented on dev-postgres-powersync/powersync-services/powersync.yaml:
```diff
@@ -44,21 +47,8 @@ services:
 
     ports:
-      - ${PS_PORT}:${PS_PORT}
+      - 8080:8080
```
why are we now hardcoding this?

rolandgeider replied:
all other services have hardcoded ports so this doen't really matter. We can also add the ps_port and add a default value if we come to need this level of configurability

Dieterbe committed:
cleanup

Dieterbe committed:
update sync rules

rolandgeider committed:
Add dockerfile for development

rolandgeider committed:
Move development file to the server repo

Building it here was not really possible due to the way docker handles
its build context

Dieterbe committed:
ivm aliases

Dieterbe committed:
fix

rolandgeider committed:
Merge branch 'master' into feature/dev-postgres-powersync

# Conflicts:
#	.gitignore
#	config/prod.env
#	dev/docker-compose.yml

rolandgeider committed:
Post merge fixes...

rolandgeider committed:
Bump ivm version

rolandgeider committed:
Add exercise tables to sync rules

rolandgeider committed:
Update sync rules

rolandgeider committed:
Configure the powersync URL

This will get sent to the client along with the token

rolandgeider committed:
Bump mongo version

rolandgeider committed:
Add missing data buckets

rolandgeider committed:
When using an android emulator use the internal ip

rolandgeider committed:
Move the initialization of the replica set to the health check

This allows us to remove the other mongo service

rolandgeider committed:
Merge branch 'master' into feature/dev-postgres-powersync

# Conflicts:
#	config/dev.env
#	config/prod.env

rolandgeider committed:
Use the "docker.io" prefix for image names for consistency

rolandgeider committed:
Option doesn't exist anymore

This would make the startup take several hours to complete

rolandgeider committed:
No need to compile pg_ivm, this is not used anymore

rolandgeider committed:
Move sync rules to streams, which finally support JOINs

rolandgeider committed:
Cleanup

rolandgeider committed:
Update sync rules

rolandgeider committed:
Merge branch 'master' into feature/dev-postgres-powersync

# Conflicts:
#	config/prod.env

rolandgeider committed:
Removed variation table

This information is now available on the exercise itself

rolandgeider committed:
Correctly read out the log and session entries

rolandgeider committed:
Cast IDs as strings

This was preventing some syncs from happening

rolandgeider committed:
Cleanup

rolandgeider committed:
Cleanup and consolidate configs

rolandgeider committed:
The pg_ivm extension is not needed anymore

rolandgeider committed:
Refactor services and dev setup

Common services can now be included instead of being duplicated on
each service. Changes to volumes, etc. can still be applied as needed.

rolandgeider committed:
Put powersync behind the reverse proxy as well

This saves us the trouble of having to expose the PS service by itself and
makes updating instances much easier.

rolandgeider committed:
Collect powersync metrics in prometheus / grafana

rolandgeider committed:
Cleanup

rolandgeider committed:
Update powersync storage user

rolandgeider committed:
Also sync the (base) routine and nutritional plan tables

rolandgeider committed:
Cleanup, we don't use the mongo backend

rolandgeider committed:
Sync the ingredient units as well

rolandgeider committed:
Sync nutrition logs as well

rolandgeider committed:
Sync gallery entries to powersync

rolandgeider committed:
Sync the rest of the nutrition tables to powersync

rolandgeider committed:
Simplify sync rules

rolandgeider committed:
Update tests

rolandgeider committed:
Consolidate JWT Keys

rolandgeider committed:
Merge branch 'master' into feature/dev-postgres-powersync

# Conflicts:
#	config/nginx.conf

rolandgeider committed:
Consolidate the db configuration into PS_DATABASE_URI

rolandgeider committed:
Add consistent logging options for the services

rolandgeider committed:
Some tweaks for the postgres service

rolandgeider committed:
Also sync the exercise comments

rolandgeider committed:
Also sync the user's profile

rolandgeider committed:
Put different rules into different sync buckets

rolandgeider committed:
Use django-allauth's tables

rolandgeider committed:
Delete the reduce entries logic, this is not needed anymore

rolandgeider committed:
Properly expose the powersync ports in prod und dev

rolandgeider committed:
Remove loading of dev settings from powersync in prod

rolandgeider committed:
Remove unused JWKS audience

rolandgeider committed:
Remove disabled profile for web service

This was committed by mistake

rolandgeider committed:
Disable telemetry

rolandgeider committed:
Sync over the new "nutrition_synced_ingredient" table
